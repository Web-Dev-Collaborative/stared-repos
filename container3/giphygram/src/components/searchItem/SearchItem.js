import React from 'react';
import { gifEntityPropType } from '../../types/giphyTypes';
import './searchItem.css';

export class SearchItem extends React.Component {
  static propTypes = {
    item: gifEntityPropType.isRequired,
  };

  render() {
    const { item } = this.props;

    return (
      <div className="w-100 mb-3 search-item">
        <a href={item.url}>
          <img
            className="w-100"
            border="0"
            src={item.images.fixed_width.url}
            alt={item.title}
            title={item.title}
          />
        </a>
      </div>
    );
  }
}
