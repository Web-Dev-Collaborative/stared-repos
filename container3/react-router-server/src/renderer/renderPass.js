import React from 'react';
import { renderToString, renderToStaticMarkup } from 'react-dom/server';
import AsyncRenderer from '../components/AsyncRenderer';
import removeDuplicateModules from '../utils/removeDuplicateModules';

const renderPass = (context, element, staticMarkup = false) => {
  context.callback = () => {
    if (context.finishedLoadingModules && !context.statesRenderPass) {
      context.statesRenderPass = true;
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    } else if (context.finishedLoadingModules && context.statesRenderPass || !context.hasModules) {
      context.renderResult = renderPass(context, element, staticMarkup);
      if (context.fetchingStates <= 0 && context.modulesLoading <= 0) {
        context.resolve({
          html: context.renderResult,
          state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
          modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
        });
      }
    }
  };

  let component = (
    <AsyncRenderer context={context}>
      {element}
    </AsyncRenderer>
  );

  let result;
  try {
    if (staticMarkup) {
      result = renderToStaticMarkup(component);
    } else {
      result = renderToString(component);
    }
  } catch (e) {
    return context.reject(e);
  }

  if (!context.hasModules && !context.hasStates) {
    context.resolve({
      html: result,
      state: context.fetchStateResults === undefined ? null : context.fetchStateResults,
      modules: context.modules === undefined ? null : removeDuplicateModules(context.modules)
    });
  }
  return result;
};

export default renderPass;
