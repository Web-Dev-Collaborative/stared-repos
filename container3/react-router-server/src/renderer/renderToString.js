import renderPass from './renderPass';
import { clear } from '../module/cache';

export default (element) => {
  return new Promise((resolve, reject) => {
    clear();
    const context = {
      resolve, reject,
      modulesLoading: 0,
      fetchingStates: 0
    };
    renderPass(context, element);
  });
};
