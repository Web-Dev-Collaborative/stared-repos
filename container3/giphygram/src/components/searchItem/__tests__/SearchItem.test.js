import React from 'react';
import renderer from 'react-test-renderer';
import { SearchItem } from '../SearchItem';
import searchResultsMock from '../../../mocks/searchResults';

describe('SearchItem', () => {
  it('should be rendered correctly', () => {
    const gifEntity = searchResultsMock.data[0];
    const tree = renderer
      .create(<SearchItem item={gifEntity} />)
      .toJSON();
    expect(tree).toMatchSnapshot();
  });
});
