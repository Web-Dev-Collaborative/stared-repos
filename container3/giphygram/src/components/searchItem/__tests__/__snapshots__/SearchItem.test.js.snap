// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SearchItem should be rendered correctly 1`] = `
<div
  className="w-100 mb-3 search-item"
>
  <a
    href="https://giphy.com/gifs/k-dancing-3o6UBlHJQT19wSgJQk"
  >
    <img
      alt="k GIF"
      border="0"
      className="w-100"
      src="https://media0.giphy.com/media/3o6UBlHJQT19wSgJQk/200w.gif"
      title="k GIF"
    />
  </a>
</div>
`;
